repos/system_upgrade/el7toel8/actors/checkkdegnome/tests/unit_test.py

```diff
@@ -0,0 +1,175 @@
+
+    library.check_kde_gnome()
+
+    assert reporting.create_report.called == 0
+
+
+def test_only_gnome_with_active_apps(monkeypatch):
+    """
+    Scenatio: no KDE desktop is installed, but there are some KDE apps, which are actively used
+    Expected behavior: Report is made (severity = MEDIUM)
+
+    """
+    monkeypatch.setattr(library, "is_executable", is_executable_only_gnome)
+    monkeypatch.setattr(library, "is_installed", lambda x: True)
+    monkeypatch.setattr(library, "check_app_in_use", check_app_in_use_mocked)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    library.check_kde_gnome()
+
+    assert reporting.create_report.called == 1
+    #  Do not even touch the string! It is ugly, but because of max line lenght it has to
+    #  be written on two lines.
+    assert reporting.create_report.report_fields["title"] == "Upgrade can be\
+ performed, but KDE apps will be uninstalled."
+
+
+def test_only_kde(monkeypatch):
+    """
+    Scenario: There is only KDE installed.
+    Expected behavior: Report is generated with HIGH severity and inhibitor tag.
+    """
+    monkeypatch.setattr(library, "is_executable", is_executable_only_kde)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    library.check_kde_gnome()
+
+    assert reporting.create_report.called == 1
+    assert "inhibitor" in reporting.create_report.report_fields["flags"]
+
+
+def test_both_gnome_main_without_active_apps(monkeypatch):
+    """
+    Scenario: There are both KDE and GNOME installed, GNOME is the main desktop
+    and there are no KDE apps
+    Expected behavior: No report is generated.
+    """
+    monkeypatch.setattr(library, "is_executable", lambda x: True)
+    monkeypatch.setattr(library, "is_installed", lambda x: False)
+    monkeypatch.setattr(library, "get_xsession", get_xsession_gnome)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    library.check_kde_gnome()
+
+    assert reporting.create_report.called == 0
+
+
+def test_both_gnome_main_with_active_apps(monkeypatch):
+    """
+    Scenario: There are both KDE and GNOME installed, GNOME is the main desktop
+    and there are some KDE apps
+    Expected behavior: Report is generated about deleting KDE apps
+    """
+    monkeypatch.setattr(library, "is_executable", lambda x: True)
+    monkeypatch.setattr(library, "is_installed", lambda x: True)
+    monkeypatch.setattr(library, "get_xsession", get_xsession_gnome)
+    monkeypatch.setattr(library, "check_app_in_use", check_app_in_use_mocked)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    library.check_kde_gnome()
+
+    assert reporting.create_report.called == 1
+    #  Do not even touch the string! It is ugly, but because of max line lenght it has to
+    #  be written on two lines.
+    assert reporting.create_report.report_fields["title"] == "Upgrade can be\
+ performed, but KDE apps will be uninstalled."
```

Comment:
Pleases, do not use line wrap like this. Use rather like this:
```python
title_field = "Upgrade can be performed, but KDE apps will be uninstalled."
assert reporting.create_report.report_fields["title"] == title_field
```

This is on more places so change it please everywhere.

(just quick review; I will finish it next week;)

Reply:
No doubt this is much more better. Changed.